source/user/dft.rst

```diff
@@ -6,495 +6,145 @@ Density functional theory (DFT)
+
+.. _user_dft_custom_func:
+
+Customizing *xc* functionals
+============================
+
+The *xc* functional of choice can be customized. The simplest way to customize to achieve this is to assign a string expression to the ``DFT.xc`` attribute:
+
+  >>> HF_X, LDA_X = .6, .08
+  >>> B88_X = 1. - HF_X - LDA_X
+  >>> LYP_C = .81
+  >>> VWN_C = 1. - LYP_C
+  >>> mf_hf.xc = f'{HF_X:} * HF + {LDA_X:} * LDA + {B88_X:} * B88, {LYP_C:} * LYP + {VWN_C:} * VWN'
+  >>> mf_hf.kernel()
+  >>> mf_hf.xc = 'hf'
+  >>> mf_hf.kernel()
+
+The XC functional string is parsed against a set of rules, as described below.
+
+* The given functional description must be a one-line string
+
+* The functional description is case-insensitive
+
+* The functional description string has two parts, separated by a ``,``.  The first part describes the exchange functional, the second part sets the correlation functional (as for :ref:`aliases <user_dft_predef_func>`) 
+
+  - If a ``","`` does not appear in the string, the entire string is treated as the name of a compound functional (containing both the exchange and the correlation
+    functional) which should be in the list of functional aliases. Again, if the string is not found in the aliased functional list, it is treated as an exchange functional
+
+  - To input only an exchange functional (without a correlation functional), one should leave the second part blank. E.g., ``slater,`` implies a functional with the LDA contribution only
+
+  - Correspondingly, to neglect the contribution of the exchange functional (i.e. to just use a correlation functional), one should leave the first part blank, e.g., ``',vwn'`` means a functional with VWN only
+
+  - If a compound *xc* functional is specified, no matter whether it is in the exchange part (the string in front of the comma) or the correlation part (the string behind the comma), both exchange and correlation functionals of the compound *xc* functional will be used
+
+* The functional name can be placed in an arbitrary order.  Two names need be separated by operators ``+`` or ``-``.  Blank spaces are ignored.  NOTE the parser
+  only reads the operators ``+, -, *``, while ``/`` is not supported
+
+* A functional name can have at most one factor.  If a factor is not given, it is set to ``1``.  Compound functionals can be scaled as a unit. For example, ``.5 * b3lyp`` is equivalent to ``.1 * HF + .04 * LDA + .36 * B88, .405 * LYP + .095 * VWN``
+
+* The string ``HF`` stands for exact exchange (HF K matrix). ``HF`` can be put in the correlation functional part (after the comma). Putting ``HF`` in the correlation part is the same as putting ``HF`` in the exchange part
+
+* The special string ``RSH`` means a range-separated operator. Its format is ``RSH(alpha; beta; omega)``. Another way to input range separation is to use keywords ``SR_HF`` and ``LR_HF``, e.g., ``SR_HF(.1) * alpha_plus_beta`` and ``LR_HF(.1) * alpha`` where the number in the parenthesis is the value of ``omega``
```

Comment:
The RSH kernel in PySCF is based on the error function kernel; Yukawa kernels are not supported at present. (Although I think PySCF and libcint might already contain all the ingredients since the math is similar to the evaluation of f12 integrals.)

Reply:
Good point. Please see commit e451415

Reply:
OK